### PR TITLE
Polling force merge

### PR DESCRIPTION
**Pull request by gingerwizard**
Cleaner version of Force Merge Support 
https://github.com/elastic/rally/pull/1051

1. The change adds a `mode` parameter. This defaults to `blocking` which causes the current behaviour. 
2. If set to `polling` we ignore the `request-timeout` parameter and timeout the force_merge API call after 1s. We then poll the tasks API at an interval of `poll-period` (default 10s), until no force-merge tasks are listed.
3. Currently there is no timeout on the tasks API calls. It is possible this is susceptible to cluster pressure and may itself timeout under heavy load. However, i believe this represents a valid error condition.
4. The 1s timeout is a workaround to the force_merge API not supporting an async call and returning a task id. If multiple force_merge calls are issued in different runners in parallel, all force merges must complete to unblock. This seems low risk as consecutive calls are effectively a noop if `max-num-segments` has been reached.

### CODE REVIEW

File: docs/track.rst
Comment (dliappis) [inline docs/track.rst:527]: I was wondering if need to mention any changes to existing behavior in https://github.com/elastic/rally/blob/master/docs/migrate.rst; AFAIU the default behavior remains the same, so I think we are good here.

File: esrally/track/params.py
Comment (dliappis) [inline esrally/track/params.py:624]: Isn't the default `poll-period` supposed to be `10s`?

**Review by dliappis — COMMENTED** [review]
This is basically LGTM from me. Left one question about the default poll-period, which I think is currently `1` in the code?

File: docs/track.rst
Comment (gingerwizard) [reply docs/track.rst:527]: Yes default behaviour is identical.

File: esrally/track/params.py
Comment (gingerwizard) [reply esrally/track/params.py]: Well spotted @dliappis. Bumped to 10. This is maybe a bit conservative but the users can always configure.

Comment (dliappis) [reply esrally/track/params.py]: Maybe we could have a tiny unittest to ensure this gets checked properly?

**Review by dliappis — APPROVED** [review]
LGTM (provided CI becomes green after the latest commit).

I left a suggestion that perhaps we could benefit from a unit test checking the default `poll-period` but there's no need for another review cycle.

**Pull request merged**
